### DISCUSSION TITLE
Why my PnP- SearchBox performed is not getting reflected inside my Search Result

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2814] [Q&A]
I have the following PnP- Search :-

![image](https://user-images.githubusercontent.com/79726511/222126952-a9338357-4629-4e10-851e-c66aa82984a8.png)

and the following settings inside the PnP - Search Result:-

![image](https://user-images.githubusercontent.com/79726511/222127029-f9bc99f4-04a5-4aef-9f3a-8672dd1eff6d.png)

but currently when i do a free text search the results will not get reflected, i will always get all items/documents (even if it type a string which does not exists )

![image](https://user-images.githubusercontent.com/79726511/222127733-29593a03-cb23-4a77-9a15-8822b2beabaa.png)

any advice?

**Comment by kasperbolarsen** [comment]
And you do have a {SearchTerms} in the Query Template ?
![image](https://user-images.githubusercontent.com/20593570/222135077-d857cf4b-e8b2-4a99-9b25-d62ebd308dd7.png)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen thanks for the reply, yes i define it inaddition to other Query parameters as follow:-

![image](https://user-images.githubusercontent.com/79726511/222145962-b690574c-3319-4d37-b274-48af0b1ef719.png)


any advice?

**Reply by kasperbolarsen** [reply]
Does it work if you use localSharePointResults as the result source?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen yes it worked.. but in my case i only need to search documents that why i chose Documents as the data source

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  also when i define localSharePointResults by defualt i will not get any document >> which mean my filters will not be shown by default

**Reply by kasperbolarsen** [reply]
Lets start with a basic setup: use the localSharePointResults and a query like {searchTerms} * isDocument:1

This should give you every document in your tenant. Please verify

**Comment by wobba** [comment]
I suggest going over the tutorials at https://microsoft-search.github.io/pnp-modern-search/scenarios/ first. Once working start to change the query template to your needs. And your KQL in the `path` filter is wrong as wildcards are not supported this way. See the KQL documentation https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#wildcard-operator
